tests/sklearn/test_sklearn_autolog.py

```diff
@@ -527,6 +531,119 @@ def test_meta_estimator_fit_performs_logging_only_once():
         assert len(mlflow.search_runs([run._info.experiment_id], query)) == 0
 
 
+@pytest.mark.parametrize(
+    "estimator_class_and_space",
+    [
+        (sklearn.model_selection.GridSearchCV, {"kernel": ("linear", "rbf"), "C": [1, 5, 10]}),
+        (sklearn.model_selection.RandomizedSearchCV, {"C": uniform(loc=0, scale=4)}),
+    ],
+)
+@pytest.mark.parametrize("backend", [None, "threading", "loky"])
+def test_parameter_search_estimators_produce_expected_outputs(estimator_class_and_space, backend):
+    mlflow.sklearn.autolog()
+
+    estimator_class, search_space = estimator_class_and_space
+    svc = sklearn.svm.SVC()
+    cv_model = estimator_class(svc, search_space, n_jobs=1, return_train_score=True)
+    X, y = get_iris()
+
+    def train_cv_model():
+        if backend is None:
+            cv_model.fit(X, y)
+        else:
+            with sklearn.utils.parallel_backend(backend=backend):
+                cv_model.fit(X, y)
+
+    with mlflow.start_run() as run:
+        cv_model.fit(X, y)
+        run_id = run.info.run_id
+
+    params, metrics, tags, artifacts = get_run_data(run_id)
+    expected_cv_params = truncate_dict(stringify_dict_values(cv_model.get_params(deep=False)))
+    assert expected_cv_params.items() <= params.items()
+    assert metrics == {TRAINING_SCORE: cv_model.score(X, y)}
+    assert tags == get_expected_class_tags(cv_model)
+    assert MODEL_DIR in artifacts
+    assert "best_estimator" in artifacts
+    assert "cv_results.csv" in artifacts
+
+    client = mlflow.tracking.MlflowClient()
+    child_runs = client.search_runs(
+        run.info.experiment_id, f"tags.`mlflow.parentRunId` = '{run_id}'"
+    )
+    cv_results = pd.DataFrame.from_dict(cv_model.cv_results_)
+    # We expect to have created a child run for each point in the parameter search space
+    assert len(child_runs) == len(cv_results)
+
+    # Verify that each set of parameter search results has a corresponding MLflow run
+    # with the expected data
+    for _, result in cv_results.iterrows():
+        result_params = result.get("params", {})
+        params_search_clause = " and ".join(
+            [f"params.`{key}` = '{value}'" for key, value in result_params.items()]
+        )
+        search_filter = f"tags.`mlflow.parentRunId` = '{run_id}' and {params_search_clause}"
+        child_runs = client.search_runs(run.info.experiment_id, search_filter)
+        assert len(child_runs) == 1
+        child_run = child_runs[0]
+
+        child_params, child_metrics, child_tags, child_artifacts = get_run_data(
```

Comment:
TODO: Assert on `child_params` (should contain CV params + best estimator params)